transport/http/client.go

```diff
@@ -35,113 +61,207 @@ func WithUserAgent(ua string) ClientOption {
 	}
 }
 
-// WithTransport with client transport.
-func WithTransport(trans http.RoundTripper) ClientOption {
+// WithMiddleware with client middleware.
+func WithMiddleware(m ...middleware.Middleware) ClientOption {
 	return func(o *clientOptions) {
-		o.transport = trans
+		o.middleware = middleware.Chain(m...)
 	}
 }
 
-// WithMiddleware with client middleware.
-func WithMiddleware(m ...middleware.Middleware) ClientOption {
+// WithSchema with client schema.
+func WithSchema(schema string) ClientOption {
 	return func(o *clientOptions) {
-		o.middleware = middleware.Chain(m...)
+		o.schema = schema
 	}
 }
 
-// Client is a HTTP transport client.
-type clientOptions struct {
-	ctx          context.Context
-	timeout      time.Duration
-	userAgent    string
-	transport    http.RoundTripper
-	errorDecoder DecodeErrorFunc
-	middleware   middleware.Middleware
+// WithEndpoint with client addr.
+func WithEndpoint(endpoint string) ClientOption {
+	return func(o *clientOptions) {
+		o.endpoint = endpoint
+	}
 }
 
-// NewClient returns an HTTP client.
-func NewClient(ctx context.Context, opts ...ClientOption) (*http.Client, error) {
-	trans, err := NewTransport(ctx, opts...)
+// RequestEncode is request encoder
+type RequestEncode func(in interface{}) (contentType string, body []byte, err error)
```

Comment:
RequestEncodeFunc func(context.Context, in interface{}) (contentType string, body []byte, err error)